contentctl/objects/detection_stanza.py

```diff
@@ -0,0 +1,115 @@
+import uuid
+from typing import Any, ClassVar
+import json
+import hashlib
+
+from pydantic import BaseModel, Field, PrivateAttr, computed_field
+
+
+class DetectionStanza(BaseModel):
+    """
+    A model representing a stanza for a detection in savedsearches.conf
+    """
+    # The lines that comprise this stanza, in the order they appear in the conf
+    lines: list[str] = Field(...)
+
+    # The full name of the detection (e.g. "ESCU - My Detection - Rule")
+    name: str = Field(...)
+
+    # The metadata extracted from the stanza
+    _metadata: dict[str, Any] = PrivateAttr(default={})
```

Comment:
I know that the goal of this PR is looking for changes/versions, but this metadata object has some fields that (for the purposes of ESCU) are REQUIRED and TYPED. It is a simple object, I think it has 4 fields.
I think we should define this as a Pydantic Object and, after parsing the string into a JSON object, use that json object/dict to create a validated `MetaData(BaseModel)` object. Thoughts?

Reply:
Looking into this 👀

Reply:
Added `DetectionMetadata` model

@pyth0n1c please resolve if this looks good!

Reply:
Looks great, thanks! Resolved!